webview/src/plots/components/App.test.tsx

```diff
@@ -2122,5 +2122,91 @@ describe('App', () => {
       fireEvent(panel, clickEvent)
       expect(clickEvent.stopPropagation).toHaveBeenCalledTimes(1)
     })
+
+    describe('Smooth Plots', () => {
+      it('should send a message to save the value when a vega panel slider is interacted with', async () => {
```

Comment:
This test is failing on windows testing with:

```
FAIL src/plots/components/App.test.tsx (54.55 s)
    ● App › Vega panels › Smooth Plots › should send a message to save the value when a vega panel slider is interacted with
  
      expect(jest.fn()).toHaveBeenCalledWith(...expected)
   ERROR  run failed: command  exited (1)
  
      - Expected
      + Received
  
        Object {
      -   "payload": Object {
      -     "template\\smooth.tsv": 0.4,
      -   },
      -   "type": "update-smooth-plot-values",
      +   "type": "initialized",
        },
  
      Number of calls: 1
  
        2[139](https://github.com/iterative/vscode-dvc/actions/runs/5510498105/jobs/10044789728?pr=4220#step:7:142) |         fireEvent.change(slider as HTMLInputElement, { target: { value: 0.4 } })
        2[140](https://github.com/iterative/vscode-dvc/actions/runs/5510498105/jobs/10044789728?pr=4220#step:7:143) |
      > 2[141](https://github.com/iterative/vscode-dvc/actions/runs/5510498105/jobs/10044789728?pr=4220#step:7:144) |         expect(mockPostMessage).toHaveBeenCalledWith({
             |                                 ^
        2[142](https://github.com/iterative/vscode-dvc/actions/runs/5510498105/jobs/10044789728?pr=4220#step:7:145) |           payload: { [smoothId]: 0.4 },
        2[143](https://github.com/iterative/vscode-dvc/actions/runs/5510498105/jobs/10044789728?pr=4220#step:7:146) |           type: MessageFromWebviewType.SET_SMOOTH_PLOT_VALUES
        2[144](https://github.com/iterative/vscode-dvc/actions/runs/5510498105/jobs/10044789728?pr=4220#step:7:147) |         })
  
        at Object.toHaveBeenCalledWith (src/plots/components/App.test.tsx:2141:33)
```

I checked the feature on my windows machine and it was working correctly but the tests were still failing. I've tried:

* using other `fireEvent` methods like `focus` and `keydown`
* wrapping the `expect` in a `waitFor`
* using the `userEvent` module

But nothing seems to be working. Any ideas on what's the issue?

Reply:
I'll take a look

Reply:
As the failure is on Windows I would suspect that there is an unexpected conversion somewhere with the `smoothId` and that the payload doesn't match. I'll run a couple of tests

Reply:
https://github.com/iterative/vscode-dvc/pull/4253/commits/540fb7af0543a8c79d308ef83d9aeeceba0ddee4 worked. One last thing to try.

Reply:
https://github.com/iterative/vscode-dvc/actions/runs/5514227454/jobs/10053229194?pr=4253

changing the setup to 
```
        renderAppWithOptionalData({
          template: {
            ...withVegaPanels,
            smoothPlotValues: { [smoothId]: 0.00001 }
          }
        })
```

gives the following result: 

![image](https://github.com/iterative/vscode-dvc/assets/37993418/0f4e2bca-5f14-4902-b5d8-cfb67678b217)

☝🏻 based on the above run it looks like the `fireEvent.change` is not picked up correctly.

Reply:
if you can confirm the behaviour on Windows I am ok with removing the test.

Reply:
>  I think the failure is to do with the way that we are relying on dispatch + useEffect to send the message.

Yep, fixed with removal of that combination!